src/main/scala/org/idio/dbpedia/spotlight/stores/CustomDbpediaResourceStore.scala

```diff
@@ -100,19 +106,20 @@ class CustomDbpediaResourceStore(val pathtoFolder: String,
   * Given a list of string of types it return a list wth types ids:
   * i.e: [dbpdia:person, dbpedia:location] => [100, 392]..
   * */
-  def getTypesIds(dbpediaTypes: Array[String]): Array[Array[java.lang.Short]] = {
-    var dbpediaTypesForResource2 = new Array[Array[java.lang.Short]](1)
-    var dbpediaTypesForResource: Array[java.lang.Short] = new Array[java.lang.Short](dbpediaTypes.length)
-
-    for (i <- 0 to dbpediaTypes.length - 1) {
-      var currentType: String = dbpediaTypes(i)
-      if (!currentType.equals("")) {
-        dbpediaTypesForResource(i) = resStore.ontologyTypeStore.getOntologyTypeByName(currentType).id
+  def getTypesIds(dbpediaTypes: Array[String]): Array[java.lang.Short] = {
+    dbpediaTypes.map( ty => {
+      try Some(resStore.ontologyTypeStore.getOntologyTypeByName(ty).id)
+      catch {
+        case _ => None
       }
-    }
+    }).flatten
+  }
+
+  def setOntologyTypes(resourceID:Int, ontologyTypes:Array[String]) {
+    if (!ontologyTypes.isEmpty)){
```

Comment:
maybe we could get rid of this if ?
we would just append a list whatever it is ?